File: treeView/src/main/java/org/sqlunet/treeview/control/SimpleController.java
/*
 * Copyright (c) 2023. Bernard Bou
 */

package org.sqlunet.treeview.control;

import android.content.Context;
import android.view.View;
import android.widget.TextView;

import org.sqlunet.treeview.model.TreeNode;

import androidx.annotation.NonNull;

/**
 * Simple controller
 *
 * @author Bogdan Melnychuk on 2/11/15.
 */
public class SimpleController extends Controller<Object>
{
	/**
	 * Constructor
	 */
	public SimpleController()
	{
		super(false);
	}

	@NonNull
	@Override
	public View createNodeView(@NonNull final Context context, final TreeNode node, final Object value)
	{
		final TextView textView = new TextView(context);
		textView.setText(String.valueOf(value));
		return textView;
	}
}

File: treeView/src/main/java/org/sqlunet/treeview/control/SimpleController.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.treeview.control

import android.content.Context
import android.view.View
import android.widget.TextView
import org.sqlunet.treeview.model.TreeNode

/**
 * Simple controller
 *
 * @author Bogdan Melnychuk on 2/11/15.
 */
class SimpleController : Controller<Any?>(false) {

    override fun createNodeView(context: Context, node: TreeNode, value: Any?, minHeight: Int): View {
        val textView = TextView(context)
        if (minHeight > 0) {
            textView.setMinimumHeight(minHeight)
        }
        textView.text = value.toString()
        return textView
    }
}
